test/ClientConnectContextTest.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Amp\Socket\Test;
+
+use Amp\Dns\Record;
+use Amp\Socket\ClientConnectContext;
+use PHPUnit\Framework\TestCase;
+
+class ClientConnectContextTest extends TestCase {
+    public function bindToDataProvider() {
+        return [
+            [null],
+            ['127.0.0.1:12345']
+        ];
+    }
+
+    /**
+     * @dataProvider bindToDataProvider
+     */
+    public function testWithBindTo($bindTo) {
+        $context = new ClientConnectContext();
+        $clonedContext = $context->withBindTo($bindTo);
+
+        $this->assertNotSame($clonedContext, $context);
```

Comment:
This shouldn't be tested. If the new value is equal to the old value, the same instance might be returned.

Reply:
Or rather: there should be an initial context with a different bindTo value and then compare that with the newer context.

Reply:
> Or rather: there should be an initial context with a different bindTo value and then compare that with the newer context.

And this is exactly what's happening in this test.
1. I create an initial context with default values:
```$context = new ClientConnectContext();```
2. Then call the tested method:
```$clonedContext = $context->withBindTo($bindTo);```
3. Check that the tested method returned a different instance. (The `clone` call in the tested method will always create a new instance, hence the tested method will never return the same instance.)
```$this->assertNotSame($clonedContext, $context);```
4. Check if the initial context's `bindTo` property is untouched
```$this->assertNull($context->getBindTo());```
5. Check if the new context's `bindTo` property is set correctly
```$this->assertSame($bindTo, $clonedContext->getBindTo());```

Reply:
The fact that a new instance is returned is an implementation detail that might change at any time. If the value doesn't change, the same instance can be returned.

Reply:
Alright then, would you like me to remove these kind of assertions from every test?

Reply:
went ahead and removed them...